public_html/wp-content/mu-plugins/blocks/assets/src/sessions/edit.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * External dependencies
+ */
+import { isUndefined, pickBy, split } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+const apiFetch = wp.apiFetch;
+const { withSelect } = wp.data;
+const { Component, Fragment } = wp.element;
+const { addQueryArgs } = wp.url;
+
+/**
+ * Internal dependencies
+ */
+import SessionsBlockControls from './block-controls';
+import SessionsInspectorControls from './inspector-controls';
+import GridToolbar from '../shared/grid-layout/toolbar';
+
+const blockData = window.WordCampBlocks.sessions || {};
+
+const SESSIONS_ICON = 'list-view';
+const MAX_POSTS = 100;
+
+const ALL_POSTS_QUERY = {
+	orderby  : 'title',
+	order    : 'asc',
+	per_page : MAX_POSTS,
+	_embed   : true,
+};
+
+const ALL_TERMS_QUERY = {
+	orderby  : 'name',
+	order    : 'asc',
+	per_page : MAX_POSTS,
+};
+
+class SessionsEdit extends Component {
+	constructor( props ) {
+		super();
+
+		this.state = {
+			allSessionPosts      : null,
+			allSessionTracks     : null,
+			allSessionCategories : null,
+		};
+	}
+
+	componentWillMount() {
```

Comment:
@coreymckrill , @vedanshujain : `componentWillMount()`, `componentWillRecieveProps()`, and `componentWillUpdate()` are being deprecated in React 16.3 and removed in 17; would it be best to migrate away from them now?

https://blog.northcoders.com/react-componentwillmount-to-be-deprecated
https://github.com/WordPress/gutenberg/issues/7897#issuecomment-428242442
https://github.com/facebook/react/issues/7671

Reply:
( note that there are lots of places where we use them, so if we change one we should change them all )

Reply:
er, Gutenberg is up to 16.8.4, so they're already deprecated.

Reply:
Blerg.

Reply:
Yes, we should refactor now.

Reply:
Opened #63 to address this.